src/middlewared/middlewared/plugins/catalogs_linux/items.py

```diff
@@ -93,81 +121,106 @@ def items(self, label, options):
 
                 cached_data[train] = train_data
 
+            job.set_progress(100, 'Retrieved catalog item(s) details successfully')
             return cached_data
         elif not os.path.exists(catalog['location']):
+            job.set_progress(5, f'Cloning {label!r} catalog repository')
             self.middleware.call_sync('catalog.update_git_repository', catalog, True)
 
         if all_trains:
             # We can only safely say that the catalog is healthy if we retrieve data for all trains
             self.middleware.call_sync('alert.oneshot_delete', 'CatalogNotHealthy', label)
 
-        trains = self.get_trains(
-            catalog['location'], {
-                'alert': True,
-                'label': label,
-                'all_trains': all_trains,
-                'trains': options['trains'],
-                'retrieve_versions': options['retrieve_versions'],
-            }
-        )
+        trains = self.get_trains(job, catalog, options)
 
-        if all_trains and options['retrieve_versions']:
+        if all_trains:
             # We will only update cache if we are retrieving data of all trains for a catalog
             # which happens when we sync catalog(s) periodically or manually
-            self.middleware.call_sync('cache.put', f'catalog_{label}_train_details', trains, 86400)
+            # We cache for 90000 seconds giving system an extra 1 hour to refresh it's cache which
+            # happens after 24h - which means that for a small amount of time it's possible that user
+            # come with a case where system is trying to access cached data but it has expired and it's
+            # reading again from disk hence the extra 1 hour.
+            if options['retrieve_versions']:
```

Comment:
How much memory does this eat when we cache everything?

Reply:
@yocalebo that would depend on the size of catalog, we don't have a catalog big enough which i think would affect memory right now or in the coming year i hope. However i think we shouldn't be retrieving version(s) from this endpoint at all and i have updated the docstring to point out that it has been deprecated.
I tested with a catalog with 250 items and it took approximately 6 MB of memory to cache and we are nowhere close that figure now.